System/AutocompleteConfigResolver.php

```diff
@@ -34,7 +34,7 @@ public function __construct(ScopeConfigInterface $scopeConfig)
      */
     public function getIsEnabled(): bool
     {
-        return $this->scopeConfig->getValue(self::PATH_IS_ENABLED);
+        return (bool) $this->scopeConfig->getValue(self::PATH_IS_ENABLED);
```

Comment:
Ohh was it this that caused the 500 error.

Reply:
that's correct :( I don't know if M2 has the same concept as StoreConfigFlag like M1 (?) but this change should have the same effect

Reply:
Yes M2 also have StoreConfigFlag.

Reply:
@jonpday  It was failing because of PHP 7 feature

` function getIsEnabled(): bool` 
indicates that response type has to be Boolean type
Any thing else will throw a fatal error.

Same goes for parameter types in PHP 7